nginx/etc/nginx/nginx-ssl.conf.ctmpl

```diff
@@ -0,0 +1,102 @@
+worker_processes  1;
+
+events {
+    worker_connections  1024;
+}
+
+http {
+
+    {{ $acme_domain := env "ACME_DOMAIN" }}
+
+    {{ if service "docker-registry" }}
+    upstream docker-registry {
+        # write the address:port pairs for each healthy docker-registry node
+        {{range service "docker-registry"}}
+        server {{.Address}}:{{.Port}};
+        {{end}}
+        least_conn;
+    }{{ end }}
+
+    ## Set a variable to help us decide if we need to add the
+    ## 'Docker-Distribution-Api-Version' header.
+    ## The registry always sets this header.
+    ## In the case of nginx performing auth, the header will be unset
+    ## since nginx is auth-ing before proxying.
+    map $upstream_http_docker_distribution_api_version $docker_distribution_api_version {
+        'registry/2.0' '';
```

Comment:
Good catch. https://github.com/autopilotpattern/docker-registry/pull/1/commits/d6468fb27eb0a842e9a3842b0897773aca5d5b41 fixed this down on line 67:

```diff
-            add_header 'Docker-Distribution-Api-Version' \$docker_distribution_api_version always;
+            add_header 'Docker-Distribution-Api-Version' $docker_distribution_api_version always;
```